packages/mock-block-dock/dev/test-react-block.tsx

```diff
@@ -44,19 +68,36 @@ export const TestReactBlock: BlockComponent<AppProps> = ({ graph }) => {
   return (
     <div ref={blockRef}>
       <h1>
-        Hello {properties.name}! The id of this block is {entityId}
+        <>
+          Hello{" "}
+          {blockEntity?.properties[extractBaseUri(propertyTypes.name.$id)]}! The
+          id of this block is {blockEntity?.metadata.editionId.baseId}
+        </>
       </h1>
       <h2>Block-handled name editing</h2>
       <input
         type="text"
         placeholder="This block's entity's 'name' property"
-        value={properties.name}
+        value={
+          blockEntity?.properties[extractBaseUri(propertyTypes.name.$id)] as
+            | string
+            | undefined
+        }
         onChange={async (event) => {
+          if (!blockEntity) {
+            return;
+          }
           try {
             const { data, errors } = await graphService!.updateEntity({
               data: {
-                entityId,
-                properties: { ...properties, name: event.target.value },
+                entityId: blockEntity.metadata.editionId.baseId,
+                entityTypeId: blockEntity.metadata.entityTypeId,
+                properties: {
+                  ...blockEntity.properties,
+                  [extractBaseUri(propertyTypes.name.$id)]: event.target.value,
+                },
+                leftToRightOrder: blockEntity.linkData?.leftToRightOrder,
+                rightToLeftOrder: blockEntity.linkData?.rightToLeftOrder,
```

Comment:
> Maybe we can make updateEntity take a generic which is either an Entity or whatever type for LinkEntity, and not take these link fields if it's known to be an Entity? feels weird to have to pass these fields to updateEntity for all entities.

You don't have to pass the fields if you know it's not a link entity. In this context we _probably_ could safely assume it's not, but because we haven't improved the block's parameter typing yet, I wouldn't iterate on this right now. When we start propagating the generic from `Entity` again with type generation then I think we can introduce a `LinkEntity` subtype which affirms there's `linkData`. 

> why are they at the root level here when they are nested in an Entity – i.e. why doesn't the data object passed to updateEntity have a linkData property instead of these two at the top level?

`LinkData` contains `leftEntityId` and `rightEntityId` which are currently not mutable (as I've highlighted above or on a previous PR about being a potential issue)

As such, I would prefer if we don't pass the fields in with `...(blockEntity.linkData ?? {})` just in case an implementation was badly implemented to do a plain replace.

Reply:
Understood, thank you for the explanation